src/ashpy/callbacks/classifier.py

```diff
@@ -60,5 +63,15 @@ def _log_fn(context: ClassifierContext) -> None:
         input_tensor = context.current_batch[0]
         out_label = context.current_batch[1]
 
+        rank = tf.rank(input_tensor)
+
+        # if it is an image scale it if needed
+        if (
+            tf.equal(rank, 4)
```

Comment:
Are you sure that images will always have rank == 4? I may be mistaken, but I remember that some libraries read grayscale as [x, y] and not [x, y, c]

Reply:
We need to check this. We have the same code inside log/utils.py (function log) so we maybe need to fix both of them.

Reply:
For TensorFlow images must have always rank 3. A batch of images rank 4.

@mr-ubik opencv reads grayscale images without the depth channel (WxH) but for using them with TF we **must** add the depth channel, otherwise any tf.image* transformation won't work.

In short, this is correct